app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkSearchDialogFragment.kt

```diff
@@ -0,0 +1,314 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.bookmarks
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.content.Intent
+import android.os.Build
+import android.os.Bundle
+import android.speech.RecognizerIntent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewStub
+import android.view.accessibility.AccessibilityEvent
+import android.view.inputmethod.InputMethodManager
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.constraintlayout.widget.ConstraintProperties.BOTTOM
+import androidx.constraintlayout.widget.ConstraintProperties.PARENT_ID
+import androidx.constraintlayout.widget.ConstraintProperties.TOP
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.isVisible
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.lib.state.ext.consumeFlow
+import mozilla.components.lib.state.ext.consumeFrom
+import mozilla.components.support.base.feature.UserInteractionHandler
+import mozilla.components.support.ktx.android.view.hideKeyboard
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.toolbar.ToolbarPosition
+import org.mozilla.fenix.databinding.FragmentBookmarkSearchDialogBinding
+import org.mozilla.fenix.databinding.SearchSuggestionsHintBinding
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.library.bookmarks.awesomebar.AwesomeBarView
+import org.mozilla.fenix.library.bookmarks.toolbar.ToolbarView
+import org.mozilla.fenix.settings.SupportUtils
+
+@Suppress("TooManyFunctions", "LargeClass")
+class BookmarkSearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
```

Comment:
Drive by comment: This seems to be a duplicate class to `HistorySearchDialogFragment`.  Which also duplicates parts of `SearchDialogFragment`.
To minimize the maintenance costs and prevent situations in which one bug would be resolved for just one searchFragment but not for others could the general functionality be moved to an abstract `SearchDialogFragment` which then can be extended for bookmarks / history / etc?
(Refactoring which can be left for another ticket)

Reply:
Agreed.  This is temporary.  We have a [rework](https://mozilla-hub.atlassian.net/browse/FNXV2-19758.) coming which will unified everything.